daemon/SecretsImpl/pluginwrapper.cpp

```diff
@@ -561,9 +571,22 @@ EncryptionPlugin::EncryptionAlgorithm EncryptedStoragePluginWrapper::encryptionA
 }
 
 Result EncryptedStoragePluginWrapper::collectionNames(
-        QStringList *names) const
+        QVariantMap *names) const
 {
-    return m_encryptedStoragePlugin->collectionNames(names);
+    QStringList cnames;
+    Result result = m_encryptedStoragePlugin->collectionNames(&cnames);
+    if (result.code() == Result::Succeeded) {
+        for (const QString &cname : cnames) {
+            bool locked = false;
+            Result lockedResult = m_encryptedStoragePlugin->isCollectionLocked(cname, &locked);
+            if (lockedResult.code() != Result::Succeeded) {
+                // assume locked, otherwise ignore the error.
```

Comment:
Is it wise to assume that any error here means that it's locked? I would suggest to check the error code and if there is an error here, return that error code to the caller.

Reply:
It is less useful to fail the entire operation if an attempt to read the lock state of a single collection fails, than to return at least the collection names (which have, at this point, already been retrieved without any error).  Assuming that the collection is locked is the "safe" option then.

Reply:
In this case I suggest 3 states: locked ok, unlocked ok, errored --- the 3rd state telling the client that there was an error trying to determine the state of this collection, thus it probably can't be used.